src/cryptonote_core/tx_pool.cpp

```diff
@@ -664,24 +665,32 @@ namespace cryptonote
       txi.max_used_block_id_hash = epee::string_tools::pod_to_hex(meta.max_used_block_id);
       txi.last_failed_height = meta.last_failed_height;
       txi.last_failed_id_hash = epee::string_tools::pod_to_hex(meta.last_failed_id);
-      txi.receive_time = meta.receive_time;
+      // In restricted mode we do not include this data:
+      txi.receive_time = include_sensitive_data ? meta.receive_time : 0;
       txi.relayed = meta.relayed;
-      txi.last_relayed_time = meta.last_relayed_time;
+      // In restricted mode we do not include this data:
+      txi.last_relayed_time = include_sensitive_data ? meta.last_relayed_time : 0;
       txi.do_not_relay = meta.do_not_relay;
       tx_infos.push_back(txi);
       return true;
-    }, true);
+    }, true, include_sensitive_data);
 
+    txpool_tx_meta_t meta;
     for (const key_images_container::value_type& kee : m_spent_key_images) {
       const crypto::key_image& k_image = kee.first;
       const std::unordered_set<crypto::hash>& kei_image_set = kee.second;
       spent_key_image_info ki;
       ki.id_hash = epee::string_tools::pod_to_hex(k_image);
       for (const crypto::hash& tx_id_hash : kei_image_set)
       {
-        ki.txs_hashes.push_back(epee::string_tools::pod_to_hex(tx_id_hash));
+        meta = m_blockchain.get_txpool_tx_meta(tx_id_hash);
+        // Do not include that transaction if it is not relayed and we're in restricted mode
+        if (include_sensitive_data || meta.relayed)
+          ki.txs_hashes.push_back(epee::string_tools::pod_to_hex(tx_id_hash));
```

Comment:
The DB read doesn't need to be done if include_sensitive_data. Not sure whether it really matters in practice. And you start needing a try/catch for the DB read errors now (unless the caller(s) already ha{s,ve} one).

Reply:
ok